app/controllers/LiabilityCalculationController.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.predicates.AuthorisedAction
+import javax.inject.Inject
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import services.LiabilityCalculationService
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import utils.ControllerHelper._
+
+import scala.concurrent.ExecutionContext
+
+class LiabilityCalculationController @Inject()(liabilityCalculationService: LiabilityCalculationService,
+                                               cc: ControllerComponents,
+                                               authorisedAction: AuthorisedAction)
+                                              (implicit ec: ExecutionContext) extends BackendController(cc) {
+
+  def calculateLiability(nino: String, taxYear: String, crystallisation: Option[Boolean],
+                         mtditid: String): Action[AnyContent] = authorisedAction.async(mtditid) { implicit user =>
+    liabilityCalculationService.calculateLiability(nino, taxYear, crystallisation.getOrElse(false)).map {
+      case Right(value) => Ok(value.incomeSourceId)
+      case Left(error) => handleDesErrorResponse(error.status, error.desBody)
```

Comment:
I think this can be a copy from dividends 
`case Left(errorModel) => Status(errorModel.status)(Json.toJson(errorModel.desBody))`